src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int FIGURE_COUNT = Figures.values().length;
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURE_COUNT);
+        Figure figure;
+        switch (index) {
+            case 0:
+                figure = new Square(random.nextInt(100),colorSupplier.getRandomColor());
+                break;
+            case 1:
+                figure = new Rectangle(random.nextInt(100), random.nextInt(100),
```

Comment:
It is a good practice to make all number and string values in your code constants to make it more informative